src/04_language.md

```diff
@@ -261,13 +261,53 @@ Beside panics, secure-critical drop should be protected.
 
 ### `Send` and `Sync` traits
 
-The `Send` and `Sync` traits are marker traits that allow implementors
-to specify respectively that a value can be sent (moved) to another thread and
-that a value be shared through a shared reference. They both are unsafe traits,
-that should be used. Moreover, the `unsafe` keyword is necessary for a type
-to implement those traits.
+The `Send` and `Sync` traits (defined in `std::marker` or `core::marker`) are
+marker traits used to ensure the safety of concurrency in Rust. When implemented
+correctly, they allow the Rust compiler to guarantee the absence of data races.
+Their semantics is as follows:
+
+- A type is `Send` if it is safe to send (move) it to another thread.
+- A type is `Sync` if it is safe to share a immutable reference to it with
```

Comment:
Avoid using "immutable reference" here; immutability by itself prevents data races, so an immutable object is always `Sync` and an immutable reference always `Send`. _The key to understand `Sync` is precisely to realize that `&T` is not always an immutable reference_.

  - A type `T` is `Send` if a unique access to a `T` (_e.g._, `T`, `&mut T`, `Box<T>`) is safe to send (move) to another thread.

  - A type `T` is `Sync` if a **shared** access to a `T` (_e.g._, `&T`, `Arc<T>`) is safe to send to another another thread.

      - Corollary: since a global variable (_i.e._, a `static`) is accessible from all threads, to be able to declare a `static` variable of type `T`, `T` needs to be `Sync`.

      - Counter-example: a type can offer "Interior Mutability", _i.e._, the possibility to be mutated through a _shared_ reference; then such type cannot be `Sync` unless the mutation is properly synchronized (_e.g._, using atomic operations or locks).

For more info, see https://danielhenrymantilla.github.io/posts/2019-03-03-mutation-part-3-interior-mutability/